qiskit/qobj/qasm_qobj.py

```diff
@@ -0,0 +1,482 @@
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+# pylint: disable=invalid-name
+
+"""Module providing definitions of QASM Qobj classes."""
+
+import os
+from types import SimpleNamespace
+
+import json
+import fastjsonschema
+
+
+path_part = 'schemas/qobj_schema.json'
+path = os.path.join(
+    os.path.dirname(os.path.dirname(os.path.abspath(__file__))),
+    path_part)
+with open(path, 'r') as fd:
+    json_schema = json.loads(fd.read())
+validator = fastjsonschema.compile(json_schema)
+
+
+class QasmQobjInstruction:
+    """A class representing a single instruction in an QasmQobj Experiment."""
+
+    def __init__(self, name, params=None, qubits=None, register=None,
+                 memory=None, condition=None, conditional=None, label=None,
+                 mask=None, relation=None, val=None):
+        """Instatiate a new QasmQobjInstruction object.
+
+        Args:
+            name (str): The name of the instruction
+            params (list): The list of parameters for the gate
+            qubits (list): A list of `int`s representing the qubits the
+                instruction operates on
+            register (list): If a ``measure`` instruction this is a list
+                of `int`s containing the list of register slots in which to
+                store the measurement results (must be the same length as
+                qubits). If a ``bfunc`` instruction this is a single `int`
+                of the register slot in which to store the result.
+            memory (list): If a ``measure`` instruction this is a list
+                of `int`s containing the list of memory slots to store the
+                measurement results in (must be the same length as qubits).
+                If a ``bfunc`` instruction this is a single `int` of the
+                memory slot to store the boolean function result in.
+            condition (tuple): A tuple of the form ``(int, int)`` where the
+                first `int` is the control register and the second `int` is
+                the control value if the gate has a condition.
+            conditional (int):  The register index of the condition
+            label (str): An optional label assigned to the instruction
+            mask (int): For a ``bfunc`` instruction the hex value which is
+                applied as an ``AND`` to the register bits.
+            relation (str): Relational  operator  for  comparing  the  masked
+                register to the `val` kwarg. Can be either ``==`` (equals) or
+                ``!=`` (not equals).
+            val (int): Value to which to compare the masked register. In other
+                words, the output of the function is (``register AND mask)
+                relation val.
+        """
+        super(QasmQobjInstruction, self).__init__()
```

Comment:
why is this super() needed?

Reply:
It's not, fixed in: https://github.com/Qiskit/qiskit-terra/pull/3383/commits/e10bbc951178f497260a959d900a0d4fbeae895e